### PR TITLE
Adds cookie so email capture dialog is not seen for 90 days after being closed

### PR DESCRIPTION
**Pull request by hannahbergam**
This sets up a cookie so the dialog doesn't display if the user has already seen and closed it in the past 90 days. I followed the logic of the signInCallout (in regards to setting the cookie onClose). I am very open to be swayed in a different direction- I also considered setting the cookie immediately upon rendering.

I'm having a hard time writing a unit test to mount this component (the error has to do with focus-trap) that I believe is related to the AccessibleDialog. I can't find an example of another test that mounts the AccessibleDialog. That being said, I was able to test locally for a signed in user that the dialog now only appears once. 

I can also see that the cookie was correctly set in the developer tools:
Screenshot 2023-11-08 at 10 58 31 AM

## Links

- spec: https://docs.google.com/document/d/1jSmRkX8O-PMDj9y-hvMubOMl1hkcRz8o9_nUO1s7gls/edit?pli=1
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-1142

## Testing story

I added a test to the test file to ensure the cookie is being set properly when the dialog is closed.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/dance/DanceVisualizationColumn.jsx [hunk 1]
```diff
@@ -113,9 +114,10 @@ class DanceVisualizationColumn extends React.Component {
         {!this.props.isShareView && (
           <AgeDialog turnOffFilter={this.turnFilterOff} />
         )}
-        {(this.props.over21 || this.props.userType === 'teacher') && (
-          <HourOfCodeGuideEmailDialog isSignedIn={isSignedIn} />
-        )}
+        {(this.props.over21 || this.props.userType === 'teacher') &&
+          !(cookies.get('HourOfCodeGuideEmailDialogSeen') === 'true') && (
```
Comment (bethanyaconnor) [inline apps/src/dance/DanceVisualizationColumn.jsx]: nit: `cookies.get('HourOfCodeGuideEmailGialogSeen') !== 'true'`

File: apps/src/dance/DanceVisualizationColumn.jsx [hunk 2]
```diff
@@ -113,9 +114,10 @@ class DanceVisualizationColumn extends React.Component {
         {!this.props.isShareView && (
           <AgeDialog turnOffFilter={this.turnFilterOff} />
         )}
-        {(this.props.over21 || this.props.userType === 'teacher') && (
-          <HourOfCodeGuideEmailDialog isSignedIn={isSignedIn} />
-        )}
+        {(this.props.over21 || this.props.userType === 'teacher') &&
```
Comment (bethanyaconnor) [inline apps/src/dance/DanceVisualizationColumn.jsx:117]: This has nothing to do with this PR but is our plan to show this to teachers who have already created an account?

**Review by bethanyaconnor — APPROVED** [review]
One nit but overall LGTM. I do worry this will make that CookieOverflow issue slightly worse but it's a small amount of data so hopefully not

File: apps/src/dance/DanceVisualizationColumn.jsx (hunk 1 above)
Comment (hannahbergam) [reply apps/src/dance/DanceVisualizationColumn.jsx]: 😂 Fixed TY

File: apps/src/dance/DanceVisualizationColumn.jsx (hunk 2 above)
Comment (hannahbergam) [reply apps/src/dance/DanceVisualizationColumn.jsx:117]: Yes- there are two different versions of the component: one for signed in teachers and the other for signed out users who've selected 21+

Comment (bethanyaconnor) [reply apps/src/dance/DanceVisualizationColumn.jsx:117]: Ah got it that makes sense!

**Pull request merged**